### DISCUSSION TITLE
x-transition:stay - is this a thing?

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#863] [4. General]
hi, firstly sorry if this is a noob question. Very new to alpine and JS in general. 

So i have an overlay (menu overlay) that needs to fade in and out when the menu opens and closes. The x-transition:enter, and the x-transition:leave add classes to animate (im using tailwind for the classnames) but the classes are removed at each stage of the transition (enter-start enter-end etc)

is there a way to have a class stay throughout the entire stage? i want to have opacity-75 stay whilst the menu is open.... 

my code is below

```
<div 
id="main-menu-overlay" 
x-show="isOpen"
x-transition:enter="transition-all ease-out duration-200"
x-transition:enter-start="opacity-0"
x-transition:enter-end="opacity-75"
x-transition:leave="transition-all ease-in duration-200"
x-transition:leave-start="opacity-75"
x-transition:leave-end="opacity-0"
class="inset-0 absolute w-full h-full bg-gray-900">
</div>

```

in my current example, once the `x-transition:enter-end `passes, the opacity turns to 100% as the class `opacity-75` is removed.

**Comment by KevinBatdorf** [comment]
I think you can just do `:class="isOpen ? 'opacity-75' : ''"`, right? 

If it's being removed then maybe add it to an inner div?
